src/main/java/frc/robot/subsystems/arm/Arm.java

```diff
@@ -28,24 +26,24 @@ public class Arm extends SubsystemBase {
             new MechanismLigament2d("Shoulder", ArmConstants.SHOULDER_LENGTH, 0)
     );
     private final MechanismLigament2d elbow = shoulder.append(
-            new MechanismLigament2d("Elbow", ArmConstants.ELBOW_LENGTH, 0, 10, new Color8Bit(Color.kPurple))
+            new MechanismLigament2d("Elbow", ArmConstants.ELBOW_LENGTH, 0)
     );
     private Command lastCommand = null;
     private Command currentCommand = null;
     private boolean changedToDefaultCommand = false;
 
-    private Arm(ArmIO io) {
-        this.io = io;
+
+    private Arm() {
+        if (Robot.isReal()) {
+            io = new ArmIOReal();
+        } else {
+            io = new ArmIOReal();
+        }
     }
 
     public static Arm getINSTANCE() {
         if (INSTANCE == null) {
-            if (Robot.isReal()) {
-                INSTANCE = new Arm(new ArmIOSim(inputs));
-
-            } else {
-                INSTANCE = new Arm(new ArmIOSim(inputs));
-            }
```

Comment:
The arm object doesn't need to know whether or not it's real. In the future (already exists in the robot template), there is going to be a static initializing function for each subsystem, to which the robot mode will be passed. So for now, the creation of the io should be done in the getInstance function.

Reply:
What happens if there are more than 2 implementations? For example, Falcon 500 and NEO swerve implementations (Robotinyo? 👀).